datalad/support/external_versions.py

```diff
@@ -252,6 +255,41 @@ def keys(self):
     def __contains__(self, item):
         return bool(self[item])
 
+    def add(self, name, func=None):
+        """Add a version checker
+
+        This method allows third-party libraries to define additional checks.
+        It will not add `name` if already exists.  If `name` exists and `func`
+        is different - it will override with a new `func`.  Added entries will
+        be included in the output of `dumps(query=True)`.
+
+        Parameters
+        ----------
+        name: str
+          Name of the check (usually a name of the Python module, or an
+          external command prefixed with "cmd:")
+        func: callable, optional
+          Function to be called to obtain version information. This should be
+          defined when checking the version of something that is not a Python
+          module or when this class's method for determining the version of a
+          Python module isn't sufficient.
+        """
+        if func:
+            func_existing = self.CUSTOM.get(name, None)
+            was_known = False
+            if func_existing and func_existing is not func:
+                lgr.debug(
+                    "Adding a new custom version checker %s for %s, "
+                    "old one: %s", func, name, func_existing)
+                was_known = name in self._versions
+            self.CUSTOM[name] = func
+            if was_known:
+                # pop and query it again right away to possibly replace with a new value
+                self._versions.pop(name)
+                _ = self[name]
+        if name not in self.INTERESTING:
+            self.INTERESTING.append(name)
```

Comment:
Shouldn't this happen only when `func` isn't specified?

Reply:
I had it first as `elif` but then decided against it since in effect it would be the same since all of those are used only to populate the known versions.  But since indeed we are not adding callable checks to INTERESTING in general, I will change it now to `elif`.

Reply:
hm, actually @mih did consider `cmd:7z` so special that now we do have it also in INTERESTING.  I think at some point we should RF it so to make INTERESTING handling independent of `CUSTOM` i.e. CUSTOM not being queried by default in `dumps`, but then here argument `interesting` would come back to make it explicit.
So, for now -- I actually think it (`if` vs `elif`) doesn't matter one way or another here , so will leave it as is to not stress CI for no good reason, unless we decide to also fix up all this INTERESTING handling and add explicitly all the `cmd:git` etc into `INTERESTING` as well.

Reply:
It doesn't matter functionally, but it's confusing with the current code.  wtf and .dumps are chaining together CUSTOM and INTERESTING, so this will lead to unnecessary repeats.  Of course these don't matter performance-wise, but it's confusing, and, given it can be avoided with an `elif`, I don't see why we shouldn't.

re: cmd:7z: I'd say that's a mistake, at least with the way the current code is written.

Reply:
ok, what do you think about (me) extend this PR separating our CUSTOM from INTERESTING (so dumps queries only INTERESTING) and returning `interesting` option here?

Reply:
I guess I'm not so bothered by the current behavior of treating CUSTOM as "interesting" by definition and not requiring them to be part of INTERESTING.  Why else would someone add a custom function if they didn't consider it interesting?  And again, with the `.add` method, I don't see when someone would go through the trouble of adding a version checker if they didn't consider it interesting.

If things are reworked, I'd consider a more useful change to be a method/property that does the chaining of INTERESTING and CUSTOM underneath.  Then wtf/.dumps and others could call it that rather than worry about which attributes contain a list of interesting things.

So, I don't know, if you feel strongly that what you're proposing is an improvement, go for it, but I'm slightly negative on the change (as well as coupling it with this PR).  I'm also okay with no change to this PR at all, though I'd prefer to see the `elif` here.

Reply:
Ok, will do `elif` and since changes to come:
yes -- I thought to add such a method as well -- will do right here.

Reply:
FWIW, avoided a new method, see d9aaae807d279d95d93f2f022541ecb2cecabd35 pushed